File: app/src/main/java/com/appsys/android/popularmovie/adapter/ReviewAdapter.java
package com.appsys.android.popularmovie.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.appsys.android.popularmovie.classes.MovieReview;
import com.shakirfattani.course.movielisting.R;

import java.util.ArrayList;

/**
 * Created by shakir on 8/2/2017.
 */

public class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.TrailerAdapterViewHolder> {

    private ArrayList<MovieReview> mMovies;

    public ReviewAdapter() {
        mMovies = new ArrayList<MovieReview>();
    }

    public void resetMovieList() {
        mMovies.clear();
        notifyDataSetChanged();
    }

    public ArrayList<MovieReview> getMoviesData() {
        return mMovies;
    }

    public void setMoviesData(ArrayList<MovieReview> movieArray) {
        mMovies.addAll(movieArray);
        notifyDataSetChanged();
    }

    @Override
    public TrailerAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.movie_reviews_item, parent, false);
        return new TrailerAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(TrailerAdapterViewHolder holder, int position) {
        holder.bind(mMovies.get(position));
    }

    @Override
    public int getItemCount() {
        return mMovies.size();
    }

    public class TrailerAdapterViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public final TextView mAuthor;
        public final TextView mReview;

        public TrailerAdapterViewHolder(View view) {
            super(view);
            mAuthor = view.findViewById(R.id.reviewer_name);
            mReview = view.findViewById(R.id.review);
            view.setOnClickListener(this);
        }

        void bind(MovieReview m) {
            mAuthor.setText(m.getAuthor());
            mReview.setText(m.getMessage());
        }

        @Override
        public void onClick(View view) {
            int adapterPosition = getAdapterPosition();
            MovieReview m = mMovies.get(adapterPosition);
        }
    }
}

File: app/src/main/java/com/appsys/android/popularmovie/adapter/ReviewAdapter.kt
package com.appsys.android.popularmovie.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.appsys.android.popularmovie.classes.MovieReview
import com.shakirfattani.course.movielisting.R
import kotlinx.android.synthetic.main.movie_reviews_item.view.*
import java.util.*

/**
 * Created by shakir on 8/2/2017.
 */
class ReviewAdapter(val mMovies: ArrayList<MovieReview> = arrayListOf()) : RecyclerView.Adapter<ReviewAdapter.TrailerAdapterViewHolder>() {
    fun resetMovieList() {
        mMovies.clear()
        notifyDataSetChanged()
    }

    var moviesData: ArrayList<MovieReview>?
        get() = mMovies
        set(movieArray) {
            mMovies.addAll(movieArray!!)
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
    =TrailerAdapterViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.movie_reviews_item, parent, false))


    override fun onBindViewHolder(holder: TrailerAdapterViewHolder, position: Int) {
        holder.bind(mMovies[position])
    }

    override fun getItemCount() = mMovies.size

    inner class TrailerAdapterViewHolder(val view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
        fun bind(m: MovieReview) {
            view.reviewer_name.text = m.author
            view.review.text = m.message
        }

        override fun onClick(view: View) {
            val adapterPosition = adapterPosition
            val m = mMovies[adapterPosition]
        }

        init {
            view.setOnClickListener(this)
        }
    }
}